airbyte-integrations/connectors/source-corebos/source_corebos/source.py

```diff
@@ -0,0 +1,130 @@
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+import requests
+import hashlib
+
+from airbyte_protocol import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from base_python import AirbyteLogger, Source
+
+
+class SourceCorebos(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        try:
+            params={"operation":"getchallenge","username":config["username"]}
+            base_url="http://test.coreboscrm.com/denorm/webservice.php"
+            url = base_url
+            logger.info("GET {}".format(url))
+            response = requests.get(url, params=params)
+            if response.status_code == 200:
+                return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+            else:
+                return AirbyteConnectionStatus(
+                    status=Status.FAILED,
+                    message=f"An exception occurred: Status Code: {0}, content: {1}".format(response.status_code, response.content),
+                )
+        except Exception as e:
+            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
+
+    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
+        streams = []
+
+        stream_name = "retrieve" 
+        json_schema = { 
+            "type": "object",
+            "properties": {
+                "success": { "type": "boolean" },
+                "result": {
+                    "type": "object",
+                        "properties": {
+                         "user_name": { "type": "string" },
+                          "message":{"type":"you will get the schema of the endpoint selected."}
+          }
+        }
+      }
+    }
+
+
+        streams.append(AirbyteStream(name=stream_name, json_schema=json_schema))
+        return AirbyteCatalog(streams=streams)
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog:json, state: Dict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
+
+        retrieve_stream = "retrieve"
+        
+        if retrieve_stream is None:
+            logger.info("No selected stream")
+            return
```

Comment:
you are defining `retrive_stream` above, are you planning to receive this variable as an argument?

Reply:
nope

Reply:
i'm we can remove the if block then